### PR TITLE
[ADD] [15.0] sale_warn_option

### PR DESCRIPTION
**Pull request by Shide**
This module extends the functionality of blocking messages to support more consistency
between messages and to allow you to set a blocking message quickly.

This is a PR of a series:

[base_warn_option](https://github.com/OCA/server-ux/pull/853)
[purchase_warn_option](https://github.com/OCA/purchase-workflow/pull/2240)
[account_warn_option](https://github.com/OCA/account-invoicing/pull/1704)
[stock_warn_option](https://github.com/OCA/stock-logistics-workflow/pull/1571)

MT-5193 @moduon @rafaelbn @yajo @EmilioPascual please review if you want :)

### CODE REVIEW

File: sale_warn_option/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+{
+    "name": "Sale Warn Option",
+    "summary": "Add Options to Sale Warn Messages",
+    "version": "15.0.1.0.0",
+    "development_status": "Alpha",
+    "category": "Sales/Sales",
+    "website": "https://github.com/OCA/sale-workflow",
+    "author": "Moduon, Odoo Community Association (OCA)",
+    "maintainers": ["Shide"],
```
Comment (yajo) [inline sale_warn_option/__manifest__.py]:
```suggestion
    "maintainers": ["Shide", "rafaelbn"],
```

See [our guidelines](https://doc.moduon.team/doc/estandares-de-contribucion-gxxTTBnNnb#h-mantenedores).

File: sale_warn_option/readme/CONTEXT.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+This module has been developed to maintain consistency in the messages used when
+blocking customers or products in sale models.
+
+If you need this module for those reasons, these might interest you too:
+
+- base_warn_option
+- sale_warn_option
```
Comment (yajo) [inline sale_warn_option/readme/CONTEXT.md]: suggestion: these 2 are already installed, so maybe you don't need to repeat them:

```suggestion
```

File: sale_warn_option/tests/test_abstract_something.py
Comment (yajo) [inline sale_warn_option/tests/test_abstract_something.py:1]: issue: this file is garbage.

File: sale_warn_option/tests/test_something.py
Comment (yajo) [inline sale_warn_option/tests/test_something.py:1]: issue: garbage too

**Review by yajo — CHANGES_REQUESTED** [review]
Looks good!

File: sale_warn_option/readme/USAGE.md
Comment (EmilioPascual) [inline sale_warn_option/readme/USAGE.md:1]: IMHO: Maybe add doc configuration, if you don't know about activate sale warnings, you don't test or use module.

File: sale_warn_option/readme/USAGE.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+To use this module, you need to:
+
+- Go to a Customer
```
Comment (EmilioPascual) [inline sale_warn_option/readme/USAGE.md:3]:
```suggestion
- Go to Sales > Orders > Customers.
```

File: sale_warn_option/readme/USAGE.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+- Select Warn Options under _Warning on the Sales Order_
```
Comment (EmilioPascual) [inline sale_warn_option/readme/USAGE.md:4]:
```suggestion
- In Internal Notes tab select Warn Options under _Warning on the Sales Order_
```

File: sale_warn_option/readme/USAGE.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+- Go to a Product
```
Comment (EmilioPascual) [inline sale_warn_option/readme/USAGE.md:5]:
```suggestion
- Go to Sales > Products > Products.
```

File: sale_warn_option/readme/USAGE.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+- Select Warn Options under _Warning when Selling this Product_
```
Comment (EmilioPascual) [inline sale_warn_option/readme/USAGE.md:6]:
```suggestion
- In a Sales tab select Warn Options under _Warning when Selling this Product_
```

File: sale_warn_option/models/product_template.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    sale_line_warn_option = fields.Many2one(
+        comodel_name="warn.option",
+    )
+
+    @api.onchange("sale_line_warn_option")
+    def _onchange_sale_line_warn_option(self):
```
Comment (EmilioPascual) [inline sale_warn_option/models/product_template.py:15]: Why not compute?

**Review by EmilioPascual — CHANGES_REQUESTED** [review]
Thanks @Shide, it's a very good module. Just documentation changes and a question why not use compute to set msg

**Review by yajo — APPROVED** [review]

**Comment by rafaelbn** [comment]
Please @Shide could you review @EmilioPascual comments with ❤️ ? Thank you! 😄

File: sale_warn_option/models/product_template.py (hunk 7 above)
Comment (Shide) [reply sale_warn_option/models/product_template.py:15]: I only want this to happen on the Interface to ensure the User sees the new message

**Review by EmilioPascual — APPROVED** [review]

**Review by rafaelbn — APPROVED** [review]
/ocabot merge patch

**Pull request merged**